### PR TITLE
Support conversion to bool for transaction

### PR DESCRIPTION
**Pull request by elsid**

### CODE REVIEW

File: include/ozo/impl/transaction.h [hunk 1]
```diff
@@ -27,6 +27,14 @@ class transaction {
...
 
+    bool has_connection() const {
+        return impl != nullptr && impl->has_connection();
+    }
+
+    operator bool() const {
+        return has_connection();
+    }
+
     friend auto& unwrap_connection(transaction& self) noexcept {
```
Comment (thed636) [inline include/ozo/impl/transaction.h]: If it possible, could you please to change unwrap_connection customisations to the structure-based  in this pull request?

```cpp
namespace ozo {

template <typename T>
struct unwrap_connection_impl<impl::transaction<T>> {
    template <typename Transaction>
    static constexpr decltype(auto) apply(Transaction&& v) noexcept {
        return unwrap_connection(*v.impl->connection);
    }
};

} // namespace zoo
```

Comment (elsid) [reply include/ozo/impl/transaction.h]: This wouldn't work as is, because `impl` is a private field. I still need to use a friend function or class.

Comment (thed636) [reply include/ozo/impl/transaction.h]: Well, in this case friendship of `unwrap_connection_impl` is preferable.

Comment (elsid) [reply include/ozo/impl/transaction.h]: Done

**Review by thed636 — APPROVED** [review]

**Pull request merged**